### PR TITLE
 Download requirementst.txt dependency locally to register it to the dependency graph

### LINKED ISSUE #1644: Workflow linter to analyse requirements.txt

### PR DESCRIPTION
**Pull request by JCZuurmond**
## Changes
Support linting job tasks with requirements.txt dependency

### Linked issues
Resolves #1644
Similar to #1704

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [x] modified existing workflow: `experimental-workflow-linter`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [x] manually tested
- [x] added unit tests
- [x] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

**Comment by JCZuurmond** [comment]
Waiting for changes to install the requirements.txt with `PipResolver`

File: src/databricks/labs/ucx/source_code/python_libraries.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def with_next_resolver(self, resolver: BaseLibraryResolver) -> PipResolver:
 
     def resolve_library(self, path_lookup: PathLookup, library: Path) -> MaybeDependency:
         library_path = self._locate_library(path_lookup, library)
-        if library_path is None:  # not installed yet
+        if library_path is None or library_path.suffix == ".txt":  # not installed yet
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/python_libraries.py:40]: Read file line by line and make multiple register library calls

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/source_code/python_libraries.py (hunk 1 above)
Comment (JCZuurmond) [reply src/databricks/labs/ucx/source_code/python_libraries.py]: done

**Comment by JCZuurmond** [comment]
> Waiting for changes to install the requirements.txt with `PipResolver`

Not required anymore after change to read `requirements.txt` line by line

File: src/databricks/labs/ucx/source_code/jobs.py [hunk 2]
```diff
@@ -99,10 +100,18 @@ def _register_library(graph: DependencyGraph, library: compute.Library) -> Itera
...
             # TODO: https://github.com/databrickslabs/ucx/issues/1640
             yield DependencyProblem("not-yet-implemented", "Wheel library is not yet implemented")
-        if library.requirements:
-            # TODO: download and add every entry via graph.register_library
-            # TODO: https://github.com/databrickslabs/ucx/issues/1644
-            yield DependencyProblem('not-yet-implemented', 'Requirements library is not yet implemented')
+        if library.requirements:  # https://pip.pypa.io/en/stable/reference/requirements-file-format/
+            logger.info(f"Registering libraries from {library.requirements}")
+            with self._ws.workspace.download(library.requirements, format=ExportFormat.AUTO) as remote_file:
+                for requirement in StringIO(remote_file.read().decode()).readlines():
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/jobs.py:106]:
```suggestion
                contents = remote_file.read().decode()
                for requirement in contents.splitlines():
```

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**